src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs

```diff
@@ -2,42 +2,31 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.ComponentModel;
 using System.ComponentModel.Design;
-using System.Diagnostics;
-using System.Globalization;
-using System.Security.Permissions;
 
 namespace System.ComponentModel
 {
     /// <summary>
-    ///    <para>Specifies the class to use to implement design-time services.</para>
+    /// Specifies the class to use to implement design-time services.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Interface, AllowMultiple = true, Inherited = true)]
     public sealed class DesignerAttribute : Attribute
     {
         private string _typeId;
 
         /// <summary>
-        ///    <para>
-        ///       Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the name of the type that
-        ///       provides design-time services.
-        ///    </para>
+        /// Initializes a new instance of the <see cref='System.ComponentModel.DesignerAttribute'/> class using the name of the type that
+        /// provides design-time services.
         /// </summary>
         public DesignerAttribute(string designerTypeName)
         {
-            string temp = designerTypeName.ToUpper(CultureInfo.InvariantCulture);
-            Debug.Assert(temp.IndexOf(".DLL") == -1, $"Came across: {designerTypeName}. Please remove the .dll extension");
-            DesignerTypeName = designerTypeName;
+            DesignerTypeName = designerTypeName ?? throw new NullReferenceException();
```

Comment:
Ugh.  I understand why you're doing this, to preserve the previous behavior, but explicitly throwing NullReferenceException like this always looks wrong to me.  What breaks if we don't validate it and simply leave off the `?? throw new NullReferenceException`? @danmosemsft, do you have an opinion here?

Reply:
I see no reason to not check for null - it's valuable because it makes throws on the problematic stack, not some time later when the field is used. But this isn't .NET Framework and we're fine with changing to ANE. I suggest this or equivalent
```DesignerTypeName = designerTypeName ?? throw new ArgumentNullExeception(nameof(designerTypeName));```
and the same in other cases.

Reply:
If you want you could  possibly go further and do null checks on designerBaseTypeName since at least some of the class assumes it's not null (eg GetHashCode).

Reply:
Fixed the NRE stuff. Maybe i'll submit another PR fixing the other NREs in GetHashCode once this is merged